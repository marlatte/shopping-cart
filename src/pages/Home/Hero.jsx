import { Link } from 'react-router-dom';
import stylishGroupUrl from '../../assets/stylish-group.png';
import css from './styles/home.module.css';

export default function Hero() {
  return (
    <section className={css.hero}>
      <div className="text-container">
        <h1>Clothes that are more than just fabric</h1>
        <p>Beautiful &apos;fits that encourage you to get creative</p>
        <Link to="/products/">Shop All Categories</Link>
      </div>
      <Link to="/products/">
        <img src={stylishGroupUrl} alt="Shop these styles and more!" />
      </Link>
    </section>
  );
}
