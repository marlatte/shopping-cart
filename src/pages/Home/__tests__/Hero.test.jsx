import { expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Hero from '../Hero';

function setup() {
  return render(<Hero />, { wrapper: BrowserRouter });
}

test('has an h1 and paragraph with proper text', () => {
  const { container } = setup();
  const p = container.querySelector('p');
  const heading = screen.getByRole('heading');

  expect(heading).toHaveTextContent(/clothes that are more than fabric/i);
  expect(p).toHaveTextContent(
    /beautiful fits that encourage you to get creative/i
  );
});

test('has a text link to all products', () => {
  setup();
  expect(
    screen.getByRole('link', { name: 'Shop All Categories' })
  ).toBeInTheDocument();
});

test('has an image link to all products', () => {
  setup();
  const link = screen.getByRole('link', {
    name: 'Shop these styles and more!',
  });
  const image = screen.getByRole('img');

  expect(link).toContainElement(image);
  expect(image).toHaveAttribute('src', '/src/assets/stylish-group.png');
});

test('matches a snapshot', () => {
  const { container } = setup();
  expect(container).toMatchSnapshot();
});
