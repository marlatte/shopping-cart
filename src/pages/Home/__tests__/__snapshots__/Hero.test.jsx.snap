// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches a snapshot 1`] = `
<div>
  <section
    class="_hero_518764"
  >
    <div
      class="_text_518764"
    >
      <h1
        class="_heroHeading_518764"
      >
        Clothes that are 
        <strong>
          more
        </strong>
         than fabric
      </h1>
      <p>
        Beautiful fits that encourage you to
         
        <strong>
          getÂ creative
        </strong>
      </p>
      <a
        class="_btn_518764"
        href="/products/"
      >
        Shop 
        <span
          class="_btnCaps_518764"
        >
          All Categories
        </span>
      </a>
    </div>
    <a
      class="_imgLink_518764"
      href="/products/"
    >
      <img
        alt="Shop these styles and more!"
        src="/src/assets/stylish-group.png"
      />
    </a>
  </section>
</div>
`;
